### PR TITLE
Initial content about containerized proxy

### PR DESCRIPTION
**Pull request by aaannz**
# Description

Documentation how to prepare containerized proxy

# Target branches

Which documentation version does this PR apply to?

- [X] Master (Default)
- [ ] Manager-4.2
- [ ] Manager-4.1
- [ ] Manager-4.0

# Links

Fixes https://github.com/SUSE/spacewalk/issues/17517

### CODE REVIEW

**Comment by aaannz** [comment]
@jcayouette Here is the PR about containerized proxy. I only integrated spacecmd reference guide as it is no brainer. But I did not yet integrate other two documents in navigation.

File: modules/installation-and-upgrade/pages/proxy-container-installation.adoc [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+In this file it is possible to modify path where proxy configuration is expected, however modifying it is strongly discouraged:
+
+- [literal]`CONFIG_DIR`=[path]`/etc/uyuni/proxy`
+
+
+
+[[installation-proxy-containers-control-plane-ssh]]
+== Set management SSH listening port on different then 22
+
+One of the containers providing {productname} Proxy services is a SSH container. This container main task is to provide bridge between {productname} Server and SSH Salt clients over isolated networks. This container however has to listen on standard SSH port 22/TCP. It is therefor required that if SSH service is used on container host its listening port must be different then 22/TCP.
```
[the remark this answers is not in the document]
Comment (aaannz) [reply modules/installation-and-upgrade/pages/proxy-container-installation.adoc]: Removed

File: modules/installation-and-upgrade/pages/proxy-container-installation.adoc [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+[[proc-installation-proxy-containers-services]]
+.Procedure: Installation Of Container Services For {productname} Proxy
+
+. Log in as `root` on the container host.
+
+. Install {productname} Proxy service package:
++
+
+----
+zypper install uyuni-proxy-systemd-services
```
[the remark this answers is not in the document]
Comment (aaannz) [reply modules/installation-and-upgrade/pages/proxy-container-installation.adoc:73]: Added requirement to assign containers module channel.

File: modules/installation-and-upgrade/pages/proxy-container-installation.adoc [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+|===
+
+
+[[installation-proxy-containers-services]]
+== Install Container Services on SLES system
+
+[NOTE]
+====
+Container host used as a base for {productname} Proxy containers needs to be first registered as a salt client to the {productname} Server.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/proxy-container-installation.adoc:45]:
```suggestion
Container host used as a base for {productname} Proxy containers needs to be first registered as a Salt client to the {productname} Server.
```

File: modules/installation-and-upgrade/pages/proxy-container-installation.adoc [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+For more information how to register salt client to the {productname} Server see xref:client-configuration:registration-overview.adoc[].
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/proxy-container-installation.adoc:47]:
```suggestion
For more information about registering Salt client to the {productname} Server, see xref:client-configuration:registration-overview.adoc[].
```

File: modules/installation-and-upgrade/pages/proxy-container-installation.adoc [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+====
+
+[NOTE]
+====
+[literal]``Containers Module`` is required to be available for container host.
+====
+
+{productname} Proxy containers are using [literal]``podman`` and [literal]``systemd`` to run and manage all proxy containers.
+
+First step is to install containers control files provided by package [literal]``uyuni-proxy-systemd-services``
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/proxy-container-installation.adoc:57]:
```suggestion
First step is to install container control files provided by package [literal]``uyuni-proxy-systemd-services``.
```

File: modules/installation-and-upgrade/pages/proxy-container-installation.adoc [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+
+[[proc-installation-proxy-containers-services]]
+.Procedure: Installation Of Container Services For {productname} Proxy
+
+. Assign [literal]``Containers Module`` software channel to the container host in the {productname}.
++
+For information how to assign software channels to the system see xref:administration:channel-management.adoc[].
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/proxy-container-installation.adoc:65]:
```suggestion
For more information about assigning software channels to the system, see xref:administration:channel-management.adoc[].
```

File: modules/installation-and-upgrade/pages/proxy-container-installation.adoc [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+[[installation-proxy-containers-customize-config]]
+== Customize {productname} Proxy configuration
+
+{productname} Proxy containers require some volumes to be mounted for long term storage.
+Paths on container host for those volumes are by default:
+
+- [path]`/var/lib/uyuni/proxy-squid-cache`
+- [path]`/var/lib/uyuni/proxy-rhn-cache`
+- [path]`/var/lib/uyuni/proxy-tftpboot`
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/proxy-container-installation.adoc:84]:
```suggestion
- [path]``/var/lib/uyuni/proxy-squid-cache``
- [path]``/var/lib/uyuni/proxy-rhn-cache``
- [path]``/var/lib/uyuni/proxy-tftpboot``
```
I'm not sure whether duple backtics are needed but we usually do it that way.

File: modules/installation-and-upgrade/pages/proxy-container-installation.adoc [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+To modify location of those paths, please edit [path]`/etc/sysconfig/uyuni-proxy-systemd-services.config`.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/proxy-container-installation.adoc:86]:
```suggestion
To modify location of those paths, edit [path]``/etc/sysconfig/uyuni-proxy-systemd-services.config``.
```

File: modules/installation-and-upgrade/pages/proxy-container-installation.adoc [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+
+In this file it is possible to add custom arguments passed to podman container pod:
+
+- [literal]`EXTRA_POD_ARGS=''`
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/proxy-container-installation.adoc:90]:
```suggestion
- [literal]``EXTRA_POD_ARGS=''``
```

File: modules/installation-and-upgrade/pages/proxy-container-installation.adoc [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+In this file it is possible to modify path where proxy configuration is expected, however modifying it is strongly discouraged:
+
+- [literal]`CONFIG_DIR`=[path]`/etc/uyuni/proxy`
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/proxy-container-installation.adoc:94]:
```suggestion
- [literal]``CONFIG_DIR`=[path]`/etc/uyuni/proxy``
```

File: modules/installation-and-upgrade/pages/proxy-container-setup.adoc [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
++
+----
+spacecmd proxy_container_config_generate_cert -- <proxy_fqdn> <oarent_fqdn> <squid_max_cache> <admin_email>
+----
+
+. Answer questions presented by script, namely {productname} credentials and CA password.
++
+This will generate file `config.zip` with configuration for the {productname} Proxy containers.
++
+For more information about [literal]`spacecmd` container proxy generation see xref:reference:spacecmd/proxy_container.adoc[].
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/proxy-container-setup.adoc:55]:
```suggestion
For more information about [literal]`spacecmd` container proxy generation, see xref:reference:spacecmd/proxy_container.adoc[].
```

File: modules/reference/pages/spacecmd/proxy_container.adoc [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+The following spacecmd functions are available for use with proxy container configuration
+
+
+
+== proxy_container_config_generate_cert
+
+Create {productname} Proxy container configuration and generate new SSL server certificate for it.
+
+[source]
+--
```
Comment (keichwa) [inline modules/reference/pages/spacecmd/proxy_container.adoc:13]:
```suggestion
----
```
we usually have 4 dashes for code-like blocks.

File: modules/reference/pages/spacecmd/proxy_container.adoc [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+           Using /root/ssl-build/RHN-ORG-PRIVATE-SSL-KEY by default.
+  --ca-pass path to a file containing the password of the CA private key, will be prompted if not passed.
+  --ssl-cname alternate name of the proxy to set in the certificate. Can be provided multiple times
+  --ssl-country country code to set in the certificate. If omitted, default values from mgr-ssl-tool will be used.
+  --ssl-state state name to set in the certificate. If omitted, default values from mgr-ssl-tool will be used.
+  --ssl-city the city name to set in the certificate. If omitted, default values from mgr-ssl-tool will be used.
+  --ssl-org the organization name to set in the certificate. If omitted, default values from mgr-ssl-tool will be used.
+  --ssl-org-unit the organization unit name to set in the certificate. If omitted, default values from mgr-ssl-tool will be used.
+  --ssl-email the email to set in the certificate. If omitted, default values from mgr-ssl-tool will be used.
+--
```
Comment (keichwa) [inline modules/reference/pages/spacecmd/proxy_container.adoc:37]:
```suggestion
----
```

File: modules/reference/pages/spacecmd/proxy_container.adoc [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+
+== proxy_container_config
+
+Create {productname} Proxy container configuration and use already existing SSL server certificate.
+
+[source]
+--
```
Comment (keichwa) [inline modules/reference/pages/spacecmd/proxy_container.adoc:45]:
```suggestion
----
```

File: modules/reference/pages/spacecmd/proxy_container.adoc [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+  CA          path to the root CA used to sign the proxy certificate in PEM format
+  CRT         path to the proxy certificate in PEM format
+  KEY         path to the proxy certificate private key in PEM format
+
+options:
+  -o, --output Path where to create the generated configuration. Default: 'config.zip'
+  -p, --ssh-port SSH port the proxy listens one. Default: 22
+  -i, --intermediate-ca  Path to an intermediate CA used to sign the proxy
+            certicate in PEM format. May be provided multiple times.
+--
```
Comment (keichwa) [inline modules/reference/pages/spacecmd/proxy_container.adoc:62]:
```suggestion
----
```

**Review by keichwa — APPROVED** [review]
Just cosmetic suggestions.  Thanks for providing the documentation!

**Review by keichwa — APPROVED** [review]
If no more feedback arrives, or if nobody vetoes, I'll merge it tomorrow (or earlier if wanted).

**Comment by aaannz** [comment]
We still need to decide where to connect these to the navigation tree. I did modify nav pages (except reference guide which is obvious)

**Comment by jcayouette** [comment]
@aaannz I just saw your comments on connecting these documents to the nav tree. My bad I missed this during my review. Where do we want to place them?

**Pull request merged**